samples/subsys/fs/littlefs_blk/src/main.c

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) 2021 Lukasz Majewski, DENX Software Engineering GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Sample which uses the filesystem API and SDHC driver */
+
+#include <zephyr.h>
+#include <device.h>
+#include <logging/log.h>
+#include <fs/fs.h>
+#include <fs/littlefs.h>
+#include <stdio.h>
+
+LOG_MODULE_REGISTER(main);
+
+/* Matches LFS_NAME_MAX */
+#define MAX_PATH_LEN 255
+#define TEST_FILE_SIZE 547
+
+static int lsdir(const char *path);
+struct fs_littlefs lfsfs;
+
+static uint8_t file_test_pattern[TEST_FILE_SIZE];
+
+static void incr_pattern(uint8_t *p, uint16_t size, uint8_t inc)
+{
+	int i;
+
+	for (i = 0; i < size; i += 8) {
+		p[i] += inc;
+	}
+
+	p[size - 1] += inc;
+}
+
+static void init_pattern(uint8_t *p, uint16_t size)
+{
+	uint8_t v = 0x1;
+	int i;
+
+	memset(p, 0x55, size);
+
+	for (i = 0; i < size; i += 8) {
+		p[i] = v++;
+	}
+
+	p[size - 1] = 0xAA;
+}
+
+static void print_pattern(uint8_t *p, uint16_t size)
+{
+	int i, j = size / 16, k;
+
+	for (k = 0, i = 0; k < j; i += 16, k++) {
+		LOG_PRINTK("%02x %02x %02x %02x %02x %02x %02x %02x ",
+		       p[i], p[i+1], p[i+2], p[i+3],
+		       p[i+4], p[i+5], p[i+6], p[i+7]);
+		LOG_PRINTK("%02x %02x %02x %02x %02x %02x %02x %02x\n",
+		       p[i+8], p[i+9], p[i+10], p[i+11],
+		       p[i+12], p[i+13], p[i+14], p[i+15]);
+
+		/* Mark 512B (sector) chunks of the test file */
+		if ((k + 1) % 32 == 0) {
+			LOG_PRINTK("\n");
+		}
+	}
+
+	for (; i < size; i++) {
+		LOG_PRINTK("%02x ", p[i]);
+	}
+
+	LOG_PRINTK("\n");
+}
+
+
+static int littlefs_binary_file_adj(char *fname)
+{
+	struct fs_dirent dirent;
+	struct fs_file_t file;
+	int rc;
+
+	/*
+	 * Uncomment below line to force re-creation of the test pattern
+	 * file on the littlefs FS.
+	 */
+	/* fs_unlink(fname); */
```

Comment:
I am not sure whether we should let in commented out code.

Reply:
This is to show how one can remove (and created from scratch) the file. This is just an example code, so IMHO it can be left as it is now.